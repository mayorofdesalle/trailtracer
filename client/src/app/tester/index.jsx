import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
