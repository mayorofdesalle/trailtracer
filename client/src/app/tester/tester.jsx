import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };