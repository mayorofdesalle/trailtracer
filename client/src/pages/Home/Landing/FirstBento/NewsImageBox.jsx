import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/BentoBox';

/**
 * NewsImageBox
 * @description
 * Styled bento box to display the media related to the news description.
 **/
const NewsImageBox = styled(BentoBox)`
    grid-area: NewsImageBox;
    overflow: hidden;
`;

export default NewsImageBox;