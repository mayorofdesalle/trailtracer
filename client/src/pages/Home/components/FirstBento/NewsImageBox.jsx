import styled from 'styled-components';

import BentoBox from '@components/ui/BentoBox';
import { rotateInFT } from '@components/misc/anims';

/**
 * NewsImageBox
 * @description
 * This is a styled bento box that is used to display the media related to the news description.
 **/
const NewsImageBox = styled(BentoBox)`
    background-color: ${({ theme }) => theme.colors.primary};
    overflow: hidden;
    animation: ${rotateInFT} 1s ease-in-out;
    z-index: -1;
`;

export default NewsImageBox;