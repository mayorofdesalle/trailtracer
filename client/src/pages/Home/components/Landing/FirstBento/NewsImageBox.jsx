import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/BentoBox';

/**
 * NewsImageBox
 * @description
 * This is a styled bento box that is used to display the media related to the news description.
 **/
const NewsImageBox = styled(BentoBox)`
    grid-area: NewsImageBox;
    overflow: hidden;
`;

export default NewsImageBox;