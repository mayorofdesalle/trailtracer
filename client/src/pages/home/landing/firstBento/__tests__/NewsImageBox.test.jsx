import { describe, expect, it, render, screen } from '@app/tester';

import NewsImageBox from '../NewsImageBox';

describe('NewsImageBox', () => {

    it('should render correctly', () => {
        render(<NewsImageBox data-testid='news-image-box' />);
        const newsImageBox = screen.getByTestId('news-image-box');

        expect(newsImageBox).toBeInTheDocument();
        expect(newsImageBox).toHaveStyle('grid-area: NewsImageBox');
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<NewsImageBox />);
        expect(asFragment()).toMatchSnapshot();
    });

});