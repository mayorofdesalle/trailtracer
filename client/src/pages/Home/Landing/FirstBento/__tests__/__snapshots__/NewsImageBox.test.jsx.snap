// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewsImageBox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh sc-guDLey sc-hLQSwg jVdZA lbUchL ccoptM"
  />
</DocumentFragment>
`;
